src/components/interface/filter/FilterMenu.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Button from '@material-ui/core/Button';
+import FilterIcon from '@material-ui/icons/PlaylistAdd';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+
+import SubMenu from './SubMenu';
+
+class FilterMenu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      anchorEl: null
+    };
+  }
+
+  handleMenuOpen = (event) => {
+    this.setState({ anchorEl: event.currentTarget });
+  };
+
+  handleMenuClose = () => {
+    this.setState({
+      anchorEl: null
+    });
+  };
+
+  render() {
+    const { anchorEl } = this.state;
+    const {
+      filtered,
+      handleFilterChecked,
+      handleFilterClearAll,
+      options
+    } = this.props;
+    const anchorOrigin = { vertical: 'bottom', horizontal: 'left' };
+    const transformOrigin = { vertical: 'top', horizontal: 'left' };
+    const categories = Object.keys(options);
+    const renderOptions = (name, opt) => (opt.map(o => ({
```

Comment:
why are we defining this function inline? Would it be more useful to extract this to a function within the class? It's quite messy to keep it inline.